plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -220,14 +224,25 @@ object StmtConversionVisitor : FirVisitor<ExpEmbedding, StmtConversionContext<Re
                 val retType = implicitInvokeCall.calleeCallableSymbol.resolvedReturnType
                 args.filter { it.type is UnspecifiedFunctionTypeEmbedding }
                     .forEach {
-                        val leakFunctionObjectCall = Stmt.Assert(DuplicableFunction.toFuncApp(listOf(it.toViper())))
+                        val leakFunctionObjectCall = Stmt.Assert(
+                            DuplicableFunction.toFuncApp(
+                                listOf(it.toViper()),
+                                Position.KtSourcePosition(implicitInvokeCall.source)
```

Comment:
The position leaking into here seems like it may be annoying later; could we pass the source instead and handle it later?

Reply:
I think is not possible because should have access to `DuplicableFunction.toFuncApp`, so we have two options:

1. Add a type parameter to `funcApp`
2. Creating an extension funciton `KtSourceElement` that converts the value into a Viper's Position.

WDYT?